cmake/leveldb.cmake

```diff
@@ -80,19 +80,24 @@ target_include_directories(leveldb
 
 add_library(nowarn_leveldb_interface INTERFACE)
 if(MSVC)
-  target_compile_options(nowarn_leveldb_interface INTERFACE
-    /wd4722
-  )
   target_compile_definitions(nowarn_leveldb_interface INTERFACE
     _CRT_NONSTDC_NO_WARNINGS
   )
+endif()
+if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+  target_compile_options(nowarn_leveldb_interface INTERFACE
+    /wd4722
+  )
 else()
   try_append_cxx_flags("-Wconditional-uninitialized" TARGET nowarn_leveldb_interface SKIP_LINK
     IF_CHECK_PASSED "-Wno-conditional-uninitialized"
   )
   try_append_cxx_flags("-Wsuggest-override" TARGET nowarn_leveldb_interface SKIP_LINK
     IF_CHECK_PASSED "-Wno-suggest-override"
   )
+  try_append_cxx_flags("-Wunused-member-function" TARGET nowarn_leveldb_interface SKIP_LINK
```

Comment:
In 76659d92f1925f561736ffe0f070ed0b9fef8d23: It'd be good if commits like this could contain actual details, rather than just "Adjust diagnostic flags"; especially given this is touching non-Windows code. Also, any reason to not fix the issues in leveldb, rather than adding more suppressions?

Reply:
More details have been added to the commit message.

> Also, any reason to not fix the issues in leveldb, rather than adding more suppressions?

I'll consider this option.

Reply:
> > Also, any reason to not fix the issues in leveldb, rather than adding more suppressions?
> 
> I'll consider this option.

Please see https://github.com/bitcoin-core/leveldb-subtree/pull/53.